coffee/visitors/utilities.py

```diff
@@ -284,7 +283,9 @@ def default_retval(cls):
     """
 
     def __init__(self, symbols):
+        from coffee.utils import flatten
         self.symbols = symbols
+        self.flatten = flatten
```

Comment:
> Not sure where I should move flatten ?

`coffee.base` might work I think, although it feels a bit arbitrary. Or you could drop `flatten` altogether and use chain-star instead:

`flatten(xs)` => `list(chain(*xs))` or `list(chain.from_iterable(xs))`.

If you directly use this in a `for`, then you don't even need the `list`. Of course, you will need `from itertools import chain`.

> the other option is to re-import flatten in the member methods that use it

I think that would express the intention (of breaking circular imports) more clearly than assigning to `self`.